dotcom-rendering/src/components/Picture.tsx

```diff
@@ -290,28 +290,29 @@ export const Picture = ({
 
 	const ratio = parseInt(height, 10) / parseInt(width, 10);
 
+	/**
+	 * Immersive MainMedia elements fill the height of the viewport, meaning on mobile
+	 * devices even though the viewport width is small, we'll need a larger image to
+	 * maintain quality. To solve this problem we're using the viewport height (vh) to
+	 * calculate width, but will never go below 100vh.
+
+	 * If either of these media queries match then the browser will choose an image from the
+	 * list of sources in srcset based on the viewport list. If the media query doesn't match
+	 * it continues checking using the standard sources underneath
+	 */
+	const sizes = `${Math.max(100, Math.round(100 / ratio))}vh`;
```

Comment:
Interesting to see the convergence here. I think `100 / (height/width)` is equivalent to `100 * (width/height)`?

https://github.com/guardian/dotcom-rendering/blob/82fe9265c667bc48686c17f9a2cfe12c85be58c2/apps-rendering/src/components/MainMedia/MainMediaImage/ImmersiveMainMediaImage.tsx#L40

Why is the baseline size `100vh`? In the case of portrait images and a portrait device, wouldn't you want the height to be `100vh` and the width to be smaller than `100vh`?

Reply:
> In the case of portrait images and a portrait device, wouldn't you want the height to be `100vh` and the width to be smaller than `100vh`?

My head alway a bit small for maths, but this is related to a famous https://github.com/guardian/frontend/pull/12811.

Reply:
`sizes` here only refers to the width considered by the UA

Reply:
That makes sense, but in that case wouldn't the lower limit be `100vw` rather than `100vh` as it is here?